api/src/test/java/org/pmiops/workbench/api/DataSetControllerTest.java

```diff
@@ -307,7 +300,7 @@
     Zendesk.class,
     FreeTierBillingService.class
   })
-  static class Configuration {
+  static class Configuration extends UserServiceTestConfiguration {
```

Comment:
🤔 The problem we'll run into, though, is that Java uses single inheritance. So I can't use this same trick for the WorkspaceService or NotebookService.

Reply:
Oh, you can have multiple classes annotated with `@TestConfiguration`! So basically you can just have 
```
@TestConfiguration
static class UserServiceTestConfig extends UserServiceTestConfiguration {}
```
and do the same for the others. So this is definitely viable for the test cases.

I think what I would change would be to leave the main configuration class as `Configuration` and then add `UserServiceConfig`, `NotebookServiceConfig`, etc, as separate classes, so they're on equal footing.

Reply:
very true.  I really want this to be a Scala `trait` (mixin) but we don't have this.

Reply:
I vaguely recall it being possible to build a config class that you can `@Import` into another one, but it's 50/50 whether I made that up.

Reply:
> I vaguely recall it being possible to build a config class that you can `@Import` into another one, but it's 50/50 whether I made that up.

Definitely possible – see https://github.com/all-of-us/workbench/blob/c44a41fe2834788e9fb294c93263cf75cb406f6f/api/src/integration/java/org/pmiops/workbench/BaseIntegrationTest.java#L21-L21

Reply:
This is much better, thank you.  I'll update to this.